src/AddIns/Uno.UI.Lottie.Wasm/ts/Uno.UI.Lottie.ts

```diff
@@ -0,0 +1,255 @@
+
+			return "ok";
+		}
+
+		public static getAnimationState(elementId: number): string {
+			const animation = Lottie._runningAnimations[elementId].animation;
+
+			const state = Lottie.getStateString(animation);
+
+			return state;
+		}
+
+		private static needNewPlayerAnimation(current: LottieAnimationProperties, newProperties: LottieAnimationProperties): boolean {
+
+			if (current.jsonPath !== newProperties.jsonPath) {
+				return true;
+			}
+			if (newProperties.stretch !== current.stretch) {
+				return true;
+			}
+			if (newProperties.autoplay !== current.autoplay) {
+				return true;
+			}
+
+			return false;
+		}
+
+		private static updateProperties(
+			runningAnimation: RunningLottieAnimation,
+			newProperties: LottieAnimationProperties) {
+
+			const animation = runningAnimation.animation;
+			const runningProperties = runningAnimation.properties;
+
+			if (runningProperties == null || newProperties.rate != runningProperties.rate) {
+				animation.setSpeed(newProperties.rate);
+			}
+
+			runningAnimation.properties = newProperties;
+		}
+
+		private static createAnimation(properties: LottieAnimationProperties, animationData?: AnimationData): RunningLottieAnimation {
+			const existingAnimation = Lottie._runningAnimations[properties.elementId];
+			if (existingAnimation) {
+				// destroy any previous animation
+				existingAnimation.animation.destroy();
+				existingAnimation.animation = null;
+			}
+
+			const config = Lottie.getPlayerConfig(properties, animationData);
+			const animation = Lottie._player.loadAnimation(config);
+
+			const runningAnimation = {
+				animation: animation,
+				properties: properties
+			};
+
+			Lottie._runningAnimations[properties.elementId] = runningAnimation;
+
+			(animation as any).addEventListener("complete", (e: any) => {
+				Lottie.raiseState(animation);
+			});
+
+			(animation as any).addEventListener("loopComplete", (e: any) => {
+				Lottie.raiseState(animation);
+			});
+
+			(animation as any).addEventListener("segmentStart", (e: any) => {
+				Lottie.raiseState(animation);
+			});
+
+			(animation as any).addEventListener("data_ready", (e: any) => {
+				Lottie._numberOfFrames = animation.totalFrames;
+				Lottie.raiseState(animation);
+			});
+
+			Lottie.raiseState(animation);
+
+			return runningAnimation;
+		}
+
+		private static getStateString(animation: Lottie.AnimationItem): string {
+			const duration = animation.getDuration(false);
+
+			const state = `${animation.animationData.w}|${animation.animationData.h}|` +
+				`${animation.isLoaded}|${animation.isPaused}|${duration}`;
+			return state;
+		}
+
+		private static raiseState(animation: Lottie.AnimationItem) {
+			const element = animation.wrapper;
+			const state = Lottie.getStateString(animation);
+
+			element.dispatchEvent(new CustomEvent("lottie_state", { detail: state }));
+		}
+
+		private static getPlayerConfig(properties: LottieAnimationProperties, animationData?: AnimationData): Lottie.AnimationConfig {
```

Comment:
Codacy found an issue: [Exceeds maximum line length of 120](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)